apps/andi/lib/andi/services/url_test.ex

```diff
@@ -0,0 +1,25 @@
+defmodule Andi.Services.UrlTest do
+  @moduledoc """
+  Tests urls with a head request, returning the time to execute and status.
+  """
+  use Tesla
+
+  plug Tesla.Middleware.JSON
+
+  def test(url) do
+    case :timer.tc(&head/1, [url]) do
+      {time, {:ok, %{status: status}}} ->
+        timed_status(time, status)
+
+      {time, {:error, :nxdomain}} ->
+        timed_status(time, "Domain not found")
+
+      {time, _} ->
+        timed_status(time, "Could not complete request")
```

Comment:
If there is any possibility that timer can return anything else other than a tuple that starts with time, it'd be nice to have an even broader catch all. But if it ALWAYS returns {time, _}, than that works for me.

Reply:
Yup, that's the way the erlang library works.